examples/instance/terraform/plugin.go

```diff
@@ -343,6 +371,16 @@ func (p *plugin) Provision(spec instance.Spec) (*instance.ID, error) {
 
 	id := instance.ID(name)
 
+	storageType, storageName, storageProps, err := FindStorage(&tf)
```

Comment:
You have a point.  I suppose the intention here is a bit different from naming the `hostname`, actually.  In the hostname case, it's actually an SL-defined attribute call hostname that needs to be modified, so the logic to base it off the VMType of softlayer_virtual_guest is reasonable. 

In this case, I think we are really saying this:
*If there is a vm*, then all the accompanying resources all need to have their terraform resource names qualified by the name of the instance.  Effectively these resources like storage,etc, are in the namespace of the vm of which they are peripherals / attachments.

If we go with this logic, then what you have is very true.  I would even go so far as to say let's generalize it even more...  If there is a VM instance, all the names of all the resources will be qualified by the generated vm name (the `infrakit-xxxxx`).  

What do you think?  This way you won't even have to have all the code specific to storage, storageType, etc.